packages/embark-ui/src/components/Communication.js

```diff
@@ -30,6 +31,10 @@ class Communication extends Component {
     this.setState({
       [name]: e.target.value
     });
+
+    if(name !== "channel") return;
```

Comment:
I rarely say that, but this time you can remove the early return. Reason is, it makes more sense to do:
```
if(name === "channel") {
   this.setState({channelIsValid: e.target.value.length >= 4});
}
```
because in this case, we only set that state for that field

Reply:
You're totally right!